src/commands/ref_command_handler.js

```diff
@@ -34,22 +34,50 @@ function cppReference(message, args) {
                 }
             }
 
-
             const results = [];
             for (const ref of hrefs) {
                 if (ref.search(args[0]) > 0) {
                     console.log(ref);
                     results.push("https://en.cppreference.com/" + ref);
                 }
             }
+			
             if (results.length > 0) {
-                channel.send(results);
+                var errorMessage = "There are too many results, please use a more specific query";
+                var stringLength = 0;
+                var msg = `Results found for ${args[0]}:\n`;
+                for (var str of results) {
+                    stringLength += str.length;
+
+                    //We implement the limit INSIDE the loop, and break if we are done, so we don't have to loop
+                    //over protentially hundreds of results if we're already out of the limit
+                    if (stringLength > 2000) {
+                        msg = {embed: {
+                            color: 16525315,
+                            fields: [{
+                                name: "Error",
+                                value: errorMessage
+                            }]
+                        }};
+                        break;
+                    }
+
+                    msg += str + "\n";
+                }
+
+                channel.send(msg);
             }
             else {
-                channel.send(`I cannot find anything in C++ with ${args[0]}`);
+                channel.send({embed: {
+                    color: 16525315,
+                    fields: [{
+                        name: "Error",
+                        value: `I cannot find anything in C++ with ${args[0]}`
+                    }]
+                }});
```

Comment:
Good use of embed here!

Reply:
I think I will do what I said above, in the first comment.